vm/src/builtins/type.rs

```diff
@@ -523,6 +523,22 @@ impl PyType {
         Ok(())
     }
 
+    #[pygetset(magic)]
+    fn annotations(&self, vm: &VirtualMachine) -> PyObjectRef {
+        self.attributes
+            .read()
+            .get(identifier!(vm, __annotations__))
+            .cloned()
+            .and_then(|found| {
+                if found.fast_isinstance(vm.ctx.types.dict_type) {
+                    None
+                } else {
+                    Some(found)
+                }
+            })
+            .unwrap_or_else(|| vm.ctx.new_dict().into())
```

Comment:
I am not really getting how this is going. Is it creating a new dict when the found `__annotations__` is a dict?

Reply:
Oh. If there is no `__annotations__` inside the dict, we need to create `__annotations__` into the dict and return that created one.